redwood-treehouse-host/src/iosMain/kotlin/app/cash/redwood/treehouse/TreehouseUIView.kt

```diff
@@ -50,26 +54,41 @@ public class TreehouseUIView private constructor(
   public constructor(widgetSystem: WidgetSystem<UIView>) : this(widgetSystem, RootUiView())
 
   init {
-    (view as RootUiView).treehouseView = this
+    view.treehouseView = this
   }
 
   private fun superviewChanged() {
     readyForContentChangeListener?.onReadyForContentChanged(this)
   }
 
-  private class RootUiView : UIView(cValue { CGRectZero }) {
+  /**
+   * The root view is just a vertical stack.
+   *
+   * In practice we expect this to contain either zero child subviews (especially when
+   * newly-initialized) or one child subview, which will usually be a layout container.
+   *
+   * This could just as easily be a horizontal stack. A box would be even better, but there's no
+   * such built-in component and implementing it manually is difficult if we want to react to
+   * content resizes.
+   */
+  private class RootUiView : UIStackView(cValue { CGRectZero }) {
     lateinit var treehouseView: TreehouseUIView
 
+    init {
+      this.axis = UILayoutConstraintAxisVertical
+      this.alignment = UIStackViewAlignmentFill // Fill horizontal.
+      this.distribution = UIStackViewDistributionFillEqually // Fill vertical.
+    }
+
     override fun layoutSubviews() {
+      super.layoutSubviews()
+
       // Bounds likely changed. Report new size.
       treehouseView.updateUiConfiguration()
-
-      subviews.forEach {
-        (it as UIView).setFrame(bounds)
-      }
     }
 
     override fun didMoveToSuperview() {
+      super.didMoveToSuperview()
```

Comment:
I’m unsure if this is strictly necessary!

Reply:
it's the right thing to do in general for any `UIView` hook

Reply:
Probably not needed in this case according to [documentation](https://developer.apple.com/documentation/uikit/uiview/1622433-didmovetosuperview?language=objc).

>The default implementation of this method does nothing. Subclasses can override it to perform additional actions whenever the superview changes.